transport-classes-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java

```diff
@@ -56,9 +57,9 @@ class EpollEventLoop extends SingleThreadEventLoop {
         Epoll.ensureAvailability();
     }
 
-    private final FileDescriptor epollFd;
-    private final FileDescriptor eventFd;
-    private final FileDescriptor timerFd;
+    private FileDescriptor epollFd;
+    private FileDescriptor eventFd;
+    private FileDescriptor timerFd;
```

Comment:
What are the "threading guarantees" ?

Reply:
These fields should be always accessed only from the eventloop thread; as the javadocs on `closeFileDescriptors` says it's up to caller to ensure that noone touches the fields when closed. I could add assertion on being in eventloop into `closeFileDescriptors` (as I assumed it's called from there); `openFileDescriptors` should be, as well, though the assertion won't be there (or would  be more complicated) as the first time it is invoked from another thread (from constructor).